packages/switch/src/react/index.js

```diff
@@ -71,69 +33,84 @@ const styles = {
-    }),
+    })
+  },
   checkbox: _ => glamor.css(css['.psds-switch__checkbox'])
 }
 
 class Switch extends React.Component {
-  constructor() {
-    super()
-    this.state = { isFocused: false }
+  constructor (props) {
+    super(props)
+
     this.handleFocus = this.handleFocus.bind(this)
     this.handleBlur = this.handleBlur.bind(this)
+
+    this.state = { isFocused: false }
   }
-  handleFocus() {
-    this.setState({ isFocused: true })
+
+  handleFocus () {
+    this.setState(_ => ({ isFocused: true }))
   }
-  handleBlur() {
-    this.setState({ isFocused: false })
+
+  handleBlur () {
+    this.setState(_ => ({ isFocused: false }))
   }
-  render() {
-    const { context, props, state } = this
-    const allProps = {
-      ...props,
-      isFocused: state.isFocused,
-      themeName: context.themeName || themeDefaultName
-    }
+
+  render () {
+    const { isFocused } = this.state
+    const { children, error } = this.props
+    const themeName = this.context.themeName || themeDefaultName
+    const allProps = { ...this.props, themeName }
+
     const switchProps = {
       ...styles.switch(allProps),
-      ...(props.onClick && !props.disabled
-        ? { onClick: _ => props.onClick(!allProps.checked) }
+      ...(allProps.onClick && !allProps.disabled
+        ? { onClick: _ => allProps.onClick(!allProps.checked) }
         : null),
-      ...(props.style ? { style: props.style } : null),
-      ...(props.className ? { className: props.className } : null)
+      ...(allProps.style ? { style: allProps.style } : null),
+      ...(allProps.className ? { className: allProps.className } : null)
     }
 
     return (
       <button
         {...switchProps}
-        tabIndex={allProps.tabIndex || '0'}
-        role="checkbox"
         aria-checked={allProps.checked}
-        onFocus={this.handleFocus}
         onBlur={this.handleBlur}
+        onFocus={this.handleFocus}
+        role='checkbox'
+        tabIndex={allProps.tabIndex || '0'}
       >
-        <div {...styles.track(allProps)}>
-          <div {...styles.thumb(allProps)} />
-        </div>
-        <input
-          tabIndex="-1"
-          type="checkbox"
-          readOnly
-          checked={allProps.checked}
-          {...styles.checkbox(allProps)}
-        />
-        {allProps.children && (
-          <label {...styles.label(allProps)}>{allProps.children}</label>
-        )}
+        <Halo
+          appearance={
+            !isFocused && error
+              ? Halo.appearances.error
+              : Halo.appearances.default
+          }
+          shape={Halo.shapes.pill}
+          visible={error || isFocused}
+        >
+          <div {...styles.track(allProps)}>
+            <div {...styles.thumb(allProps)} />
+          </div>
+
+          <input
+            tabIndex='-1'
+            type='checkbox'
+            readOnly
+            checked={allProps.checked}
+            {...styles.checkbox(allProps)}
+          />
+        </Halo>
```

Comment:
nbd, but putting Halo around the div only might be a subtle indicator as to what is visible vs hidden. 🤷‍♂️